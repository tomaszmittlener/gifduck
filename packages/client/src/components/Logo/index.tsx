import React, { FunctionComponent } from 'react'
import styled from 'styled-components'

interface LogoProps {
  className?: string
}

const StyledLogo = styled.label`
  font-size: ${({ theme }) => theme.ms(8)};
  font-family: ${({ theme }) => theme.typography.fonts.secondary};
  color: ${({ theme }) => theme.colors.accent.default};
  padding: ${({ theme }) => theme.ms(2)} 0;
  width: auto;
`

const Logo: FunctionComponent<LogoProps> = ({ className }) => {
  return <StyledLogo className={className}>🐤 gifduck</StyledLogo>
}

export default Logo
