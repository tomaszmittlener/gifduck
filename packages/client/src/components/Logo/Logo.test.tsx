import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'

import Logo from './'

const defaultProps = {
  className: 'class-name',
  testId: 'test-id',
}

afterEach(cleanup)

describe('Logo', () => {
  test('renders correctly', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <Logo />
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('passes className to container', async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <Logo {...defaultProps} />
      </ThemeProvider>,
    )

    expect(getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  })
})
