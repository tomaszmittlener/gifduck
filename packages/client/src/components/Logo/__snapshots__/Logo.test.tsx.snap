// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
.c0 {
  font-size: 2.2806973457336426rem;
  font-family: Pacifico,sans-serif;
  color: #F2C94C;
  line-height: 2;
  width: auto;
  -webkit-animation: dGzoaH 1s ease-out none;
  animation: dGzoaH 1s ease-out none;
}

<div>
  <label
    class="c0"
  >
    üê§gifduck
  </label>
</div>
`;
